### ISSUE TITLE
Improve viewing images in the app

### ISSUE THREAD

**Issue by sivaraam** [commons-app/apps-android-commons#2012]
**Summary:** 
Now that the 'Explore' feature has become mainstream. I think the app is not just an uploading app anymore (Off-topic: wondering why the play store app description doesn't mention anything about 'Explore :thinking:. Is 'Explore' just in beta?)

Unfortunately viewing an image doesn't seem to be that good currently. Issues:
* Only part of the image is shown when the we tap on an image
* No way to see full-sized image

Currently, I have to download an image or open it in the browser to view it fully.

**Steps to reproduce:** 
1. Open the app
2. Tap on any uploaded image or tap on an image via 'Explore'

*Expected*: I'm able to view the complete image
*Actual*: Only part of the image is visible. Have to download to view the image completely.

**Device and Android version:** 
Samsung Galaxy J1 Ace, Android 5.1.1

**Commons app version:** 
2.8.6

**Screen-shots:** 
![image not shown well could scale](https://user-images.githubusercontent.com/12448084/48665644-6b5ea380-ead8-11e8-8561-4a7b71284609.png)
[Actual image](https://commons.wikimedia.org/wiki/File:ICICI_Bank_Logo.svg)

![screenshot_2018-11-18-02-21-57](https://user-images.githubusercontent.com/12448084/48665662-c55f6900-ead8-11e8-95ce-d904dee13e94.png)

![screenshot_2018-11-18-02-22-01](https://user-images.githubusercontent.com/12448084/48665664-c7292c80-ead8-11e8-8383-26103f661283.png)

**Would you like to work on the issue?**
May be by providing suggestions :wink:.

**Comment by neslihanturan** [comment]
Your uploads are very good though @sivaraam :) I think this request is quite meaningful since we have Explore feature now. Currently we only have thumbnails of images, thats't why resolution is not good on Media Details Activity. My suggestion is having 2 options according to internet connection (remember #746 ):
- If save data mode is active, then user again should see thumbnail, but should be able to move on image.
- If using data is ok, then we need to fetch real image on Media Details Activity clicked. And again user should be able to move around.

**Comment by sivaraam** [comment]
> Your uploads are very good though @sivaraam

Those pictures weren't uploaded by me. :-) They were pics uploaded by other commons users. Those were some pics I got from the [POTY 2017](https://commons.m.wikimedia.org/wiki/Commons:Picture_of_the_Year#2017) candidates. I just used them to test uploading in the Commons app for a bug. I've nominated them for deletion, since. They seem to have become redirects to the original files.

Some thought on your suggestions:

> If save data mode is active, then user again should see thumbnail, but should be able to move on image.

Sounds good.

> If using data is ok, then we need to fetch real image on Media Details Activity clicked. And again user should be able to move around.

I would suggest to load thumbnail in media activity and show the full sized, good quality image when a user taps the image in the media activity. Would be good on users who don't have access to high-speed connectivity all the time (I'm one of them :-)
